programs/stake/src/stake_state.rs

```diff
@@ -756,12 +756,24 @@ impl Stake {
         stake_history: &StakeHistory,
         config: &Config,
     ) -> Result<(), StakeError> {
-        // can't redelegate if stake is active.  either the stake
-        //  is freshly activated or has fully de-activated.  redelegation
-        //  implies re-activation
+        // If stake is currently active:
         if self.stake(clock.epoch, Some(stake_history), true) != 0 {
-            return Err(StakeError::TooSoonToRedelegate);
+            // If pubkey of new voter is the same as current,
+            // and we have not deactivated any stake yet, we un-deactivate stake.
```

Comment:
Let's not introduce a term. This program has too many as it is :sweat_smile: 
```suggestion
            // and we have not deactivated any stake yet, we clear deactivation
```